src/reporter/reporter.py

```diff
@@ -116,24 +131,19 @@ def notify():
                'content.', 400
 
     payload = request.json['data']
-
-    log().info('Received payload: {}'.format(payload))
-
+    
     # preprocess and validate each entity update
     for entity in payload:
```

Comment:
I suggest moving `_filter_empty_entities` here. That is, it should be the first thing we do:

    res_entity = []
    for entity in payload:
        entity = _filter_empty_entities(entity)
        if entity is None:
            continue
       res_entity.append(e) 
    ...

This way we avoid having to use `pop` in `_filter_empty_entities` which is kinda fragile---how do you know the first iterated attribute will always be the time index?! Also as a nice side effect we avoid looping through the array twice.